Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -75,8 +98,36 @@ public override void Serialize(IIonWriter writer, object item)
         {
             options.TypeAnnotator.Apply(options, writer, targetType);
             writer.StepIn(IonType.Struct);
+
+            var serializedWithGetter = new List<string>();
+            
+            // Serialize the values returned from IonPropertyGetter annotated getter methods.
+            foreach (var method in targetType.GetMethods())
+            {
+                var getMethod = (IonPropertyGetter)method.GetCustomAttribute(typeof(IonPropertyGetter));
+                
+                // A getter method should have zero parameters.
+                if (getMethod?.IonPropertyName == null || method.GetParameters().Length != 0)
+                {
+                    continue;
+                }
+
+                writer.SetFieldName(getMethod.IonPropertyName);
+                var getValue = method.Invoke(item, Array.Empty<object>());
+                ionSerializer.Serialize(writer, getValue);
+                
+                serializedWithGetter.Add(getMethod.IonPropertyName);
```

Comment:
Is this a correct assumption to give precedence to annotated methods over property and fields? Specifically when IncludeFields is specified.

Reply:
Clarified with Ian and we agreed to skip duplicates if they've already been serialized via a method.